h2o-py/h2o/frame.py

```diff
@@ -509,7 +509,10 @@ def logical_negation(self):
 
 
     def _unop(self, op):
-        return H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        ret = H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        if ret._ex._cache._names is not None:
+            ret._ex._cache._names = ["%s(%s)" % (op, name) for name in ret._ex._cache._names]
```

Comment:
I believe `types` dictionary should be rewritten too... Since it's a name => type dictionary, and names have changed. Since your pyunit didn't catch this issue, it'd be a good idea to test the correctness of the `types` dictionary in the pyunit too.